lib/rubocop/cop/lint/return_in_void_context.rb

```diff
@@ -45,7 +45,7 @@ def on_return(return_node)
 
         def method_name(return_node)
```

Comment:
I am inclined to split this into two methods, one that finds the context, and one that extracts the name from a `def` node.

Right now these two are mixed, and causes this method to have a "special meaning `nil`" mixed return type.

I would rather see something like:

```ruby
context_node = non_void_context(return_node)

return unless context_node && ...

add_offense(return_node, :keyword, format(message, method_name(context_node))
```